Src/PerformanceCollector/Unit.Tests/QuickPulse/QuickPulseTelemetryProcessorTests.cs

```diff
@@ -384,5 +386,313 @@ public void QuickPulseTelemetryProcessorFiltersOutDependencyCallsToDefaultQuickP
+            }
+
+            // out of the second 100 items we expect to see items 100 through 129 (the new quota for 30 seconds)
+            for (int i = 5; i < 35; i++)
+            {
+                Assert.AreEqual(95 + i, collectedTelemetry[i].Duration.TotalSeconds);
+            }
+        }
+
+        [TestMethod]
+        public void QuickPulseTelemetryProcessorDoesNotCollectFullExceptionTelemetryItemsOnceQuotaIsExhausted()
+        {
+            // ARRANGE
+            var accumulatorManager = new QuickPulseDataAccumulatorManager();
+            var timeProvider = new ClockMock();
+            var telemetryProcessor = new QuickPulseTelemetryProcessor(new SimpleTelemetryProcessorSpy(), timeProvider, 60, 5);
+            var instrumentationKey = "some ikey";
+            ((IQuickPulseTelemetryProcessor)telemetryProcessor).StartCollection(
+                accumulatorManager,
+                new Uri("http://microsoft.com"),
+                new TelemetryConfiguration() { InstrumentationKey = instrumentationKey });
+
+            // ACT
+            int counter = 0;
+            for (int i = 0; i < 100; i++)
+            {
+                var exception = new ExceptionTelemetry()
+                                    {
+                                        Context = { InstrumentationKey = instrumentationKey },
+                                        Message = (counter++).ToString(CultureInfo.InvariantCulture)
+                                    };
+
+                telemetryProcessor.Process(exception);
+            }
+
+            timeProvider.FastForward(TimeSpan.FromSeconds(30));
+
+            for (int i = 0; i < 100; i++)
+            {
+                var exception = new ExceptionTelemetry()
+                                    {
+                                        Context = { InstrumentationKey = instrumentationKey },
+                                        Message = (counter++).ToString(CultureInfo.InvariantCulture)
+                                    };
+
+                telemetryProcessor.Process(exception);
+            }
+
+            // ASSERT
+            var collectedTelemetry = accumulatorManager.CurrentDataAccumulator.TelemetryDocuments.ToArray().Reverse().Cast<ExceptionTelemetryDocument>().ToArray();
+
+            Assert.AreEqual(5 + 30, accumulatorManager.CurrentDataAccumulator.TelemetryDocuments.Count);
+
+            // out of the first 100 items we expect to see items 0 through 4 (the initial quota)
+            for (int i = 0; i < 5; i++)
+            {
+                Assert.AreEqual(i, int.Parse(collectedTelemetry[i].Message, CultureInfo.InvariantCulture));
+            }
+
+            // out of the second 100 items we expect to see items 100 through 129 (the new quota for 30 seconds)
+            for (int i = 5; i < 35; i++)
+            {
+                Assert.AreEqual(95 + i, int.Parse(collectedTelemetry[i].Message, CultureInfo.InvariantCulture));
+            }
```

Comment:
This test logic seems to be repeated in multiple places, would a helper function work to simplify and de-dupe the code?